### PR TITLE
Allow user to skip registration

### PR DESCRIPTION
**Pull request by chentex**
Signed-off-by: Vicente Zepeda Mas <vzepedamas@suse.com>

Added an option to skip SCC configuration, meant for air-gapped environments.

### CODE REVIEW

**Comment by chentex** [comment]
I just have a doubt about testing the UI, I added a PushButton and wanted to test for it, I don't know if it is necessary, but I could not get a way to assert it.
Here is what I was trying:
https://github.com/yast/yast-rmt/pull/44/files#diff-4569e5801a8ed2a038e14e94b683afb9R36

File: spec/rmt/wizard_scc_page_spec.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@
 
       expect(Yast::UI).to receive(:ChangeWidget).with(Id(:scc_username), :Value, config['scc']['username'])
       expect(Yast::UI).to receive(:ChangeWidget).with(Id(:scc_password), :Value, config['scc']['password'])
+      # expect(Yast::Wizard::Content).to receive(:PushButton).with(Id(:skip), Opt(:key_F11), Yast::Label.SkipButton)
```
Comment (jreidinger) [inline spec/rmt/wizard_scc_page_spec.rb]: we usually test it with setting content and with benefit of nested find of Yast::Term like
https://github.com/yast/yast-storage-ng/blob/master/test/y2storage/dialogs/proposal_test.rb#L240 ( some helpers involved, but I hope you get idea ). Ask if you have any questions.

Comment (chentex) [reply spec/rmt/wizard_scc_page_spec.rb]: Thanks @jreidinger it worked out great :+1:

File: spec/rmt/wizard_scc_page_spec.rb [hunk 2]
```diff
@@ -21,6 +21,24 @@
 Yast.import 'Wizard'
 
 describe RMT::WizardSCCPage do
+  # Convenience method to inspect the tree of terms for the UI
+  def nested_id_term(id, term)
```
Comment (Sergeykot) [inline spec/rmt/wizard_scc_page_spec.rb]: would be nice to move those methods to a separate helper file. not a  must though.

**Comment by Sergeykot** [comment]
@chentex would it be possible to identify that system has no internet connection while checking the credentials? I think another option is to try to check them and if there is no Internet connection we can ask user if he/she want to skip the credentials check. So that we can even avoid adding 'Skip' button. What do you think?

**Comment by chentex** [comment]
It would be another option, I'm fine on doing any of them.
Bu what I don't know is, which option adheres best to the best practices of `yast`?

**Comment by jreidinger** [comment]
@Sergeykot there is a method in yast2.rpm for it - https://github.com/yast/yast-yast2/blob/master/library/network/src/modules/NetworkService.rb#L280

**Comment by jreidinger** [comment]
or to be precise the method test if there is network setup and online ( network card and associated ip address v4 or v6). We do not test reachability of a external sites.

**Comment by chentex** [comment]
> or to be precise the method test if there is network setup and online ( network card and associated ip address v4 or v6). We do not test reachability of a external sites.

So as long as there is a network interface configured it will return true?

**Comment by jreidinger** [comment]
> > or to be precise the method test if there is network setup and online ( network card and associated ip address v4 or v6). We do not test reachability of a external sites.
> 
> So as long as there is a network interface configured it will return true?

configured and up. you can check it yourself it checks ip addr output and skip localhost and search for inet keyword.

**Comment by Sergeykot** [comment]
I discussed it with the team a bit and my second thought is that the user may not even know the credentials in case of offline setup. And then it is strange to show the input fields to them.

Also it can be that network interface is configured but there are some connection issues. In that case we would have to setup some timeout.

So to conclude: let's keep it simple and clear for users. Just move `Skip` button to the footer (where `Help`, `Cancel`, `Next` buttons are) and we are fine there :smile: 

@chentex @jreidinger

**Comment by chentex** [comment]
@Sergeykot changes pushed!

**Comment by chentex** [comment]
done @Sergeykot! updated the changelog also!

**Pull request merged**